crates/pet-windows-store/src/env_variables.rs

```diff
@@ -12,7 +12,7 @@ pub struct EnvVariables {
 }
 
 impl EnvVariables {
-    pub fn from(env: &dyn Environment) -> Self {
+    pub fn from(env: Arc<dyn Environment>) -> Self {
```

Comment:
Why is this change in general needed? Arc automatically dereferences into `&T`, this is adding a lot of work cloning and making reference counters instead of borrowing the environment

Reply:
Looking through the usages it seems like you don't need an arc in [many places](https://github.com/microsoft/python-environment-tools/pull/94/files#diff-87dd895a3703313803919378a455157963acfd8ea38255f226f92bb5c42ef02aR89-R90) and now have to construct it just to make the parameter happy, so I would adjust it